sensing/autoware_cuda_pointcloud_preprocessor/src/cuda_downsample_filter/cuda_voxel_grid_downsample_filter_node.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "autoware/cuda_downsample_filter/cuda_voxel_grid_downsample_filter_node.hpp"
+
+#include "autoware/cuda_pointcloud_preprocessor/memory.hpp"
+
+namespace autoware::cuda_downsample_filter
+{
+CudaVoxelGridDownsampleFilterNode::CudaVoxelGridDownsampleFilterNode(
+  const rclcpp::NodeOptions & node_options)
+: Node("cuda_voxel_grid_downsample_filter", node_options)
+{
+  // set initial parameters
+  float voxel_size_x = declare_parameter<float>("voxel_size_x");
+  float voxel_size_y = declare_parameter<float>("voxel_size_y");
+  float voxel_size_z = declare_parameter<float>("voxel_size_z");
+
+  sub_ =
+    std::make_shared<cuda_blackboard::CudaBlackboardSubscriber<cuda_blackboard::CudaPointCloud2>>(
+      *this, "~/input/pointcloud",
+      std::bind(
+        &CudaVoxelGridDownsampleFilterNode::cudaPointcloudCallback, this, std::placeholders::_1));
+
+  pub_ =
+    std::make_unique<cuda_blackboard::CudaBlackboardPublisher<cuda_blackboard::CudaPointCloud2>>(
+      *this, "~/output/pointcloud");
+
+  cuda_voxel_grid_downsample_filter_ =
+    std::make_unique<CudaVoxelGridDownsampleFilter>(voxel_size_x, voxel_size_y, voxel_size_z);
+}
+
+void CudaVoxelGridDownsampleFilterNode::cudaPointcloudCallback(
+  const cuda_blackboard::CudaPointCloud2::ConstSharedPtr msg)
+{
+  // The following only checks compatibility with xyzi
+  // (i.e., just check the first four elements of the point field are x, y, z, and intensity
+  // and don't care the rest of the fields)
+  if (!cuda_pointcloud_preprocessor::is_data_layout_compatible_with_point_xyzi(msg->fields)) {
+    RCLCPP_ERROR(
+      this->get_logger(), "Input pointcloud data layout is not compatible with PointXYZI");
```

Comment:
if it is not compatible there should be an early return

Reply:
I modified the code so that the filter can handle different types of data for the intensity field of the input pointcloud in 0c949b35d. 
While `pointcloud_preprocessor::utils::is_data_layout_compatible_with_point_xyzi` confirms that the intensity field has float values, my filter still may work with other type such as uint8_t. For this reason, I changed here from `RCLCPP_ERROR` to `RCLCPP_WARN` without an early return.